packages/jsii-java-runtime/project/src/main/java/software/amazon/jsii/JsiiEngine.java

```diff
@@ -578,6 +576,21 @@ public JsiiObjectRef createNewObject(final Object uninitializedNativeObject, fin
         return overrides.values();
     }
 
+    private Collection<String> discoverInterfaces(final Class<?> classToInspect) {
+        final Set<String> interfaces = new HashSet<>();
+        if (classToInspect.isAnnotationPresent(Jsii.class)) {
```

Comment:
Add some comments to explain this code. I am not sure I follow why if we reach an interface we don't continue to inspect the other interfaces it implements.

Also, feels like there are a few edge cases here that are not covered in the tests.

Reply:
Will do - and maybe this will help reason over the missing edge-cases (or if we're super lucky, demonstrate there are none 😅)